dbms/src/Interpreters/SyntaxAnalyzer.cpp

```diff
@@ -541,11 +565,16 @@ void collectJoinedColumns(AnalyzedJoin & analyzed_join, const ASTSelectQuery & s
     }
 }
 
-void replaceJoinedTable(const ASTTablesInSelectQueryElement* join)
+void replaceJoinedTable(const ASTTablesInSelectQueryElement * join)
 {
     if (!join || !join->table_expression)
         return;
 
+    /// TODO: Push down for CROSS JOIN is not OK [disabled]
```

Comment:
I've also disabled table to subquery rewrite logic for CROSS JOINs here cause it doesn't work properly.